master/buildbot/changes/gitpoller.py

```diff
@@ -481,37 +484,86 @@ def _process_changes(self, newRev, branch):
                 src='git',
             )
 
+    @property
+    def sshPrivateKey(self):
+        ssh_auth_warndeprecate(
+            "self.sshPrivateKey",
+            "self._git_ssh_auth.ssh_private_key",
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth.ssh_private_key
+
+    @property
+    def sshHostKey(self):
+        ssh_auth_warndeprecate(
+            "self.sshHostKey",
+            "self._git_ssh_auth.ssh_host_key",
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth.ssh_host_key
+
+    @property
+    def sshKnownHosts(self):
+        ssh_auth_warndeprecate(
+            "self.sshKnownHosts",
+            "self._git_ssh_auth.ssh_known_hosts",
+        )
+        assert self._git_ssh_auth is not None
+        return self._git_ssh_auth.ssh_known_hosts
+
     def _isSshPrivateKeyNeededForCommand(self, command):
-        commandsThatNeedKey = [
-            'fetch',
-            'ls-remote',
-        ]
-        if self.sshPrivateKey is not None and command in commandsThatNeedKey:
-            return True
-        return False
+        ssh_auth_warndeprecate(
```

Comment:
Might have gone a bit overboard with deprecation messages...
Maybe we could remove at least private method without warnings?

Reply:
I kind of want to create mixin classes that implement these deprecation so the "real" class implementation is not "polluted".

Something like:
```python
class _DeprecatedGitStepMixin:
    @property
    def sshPrivateKey(self):
        ssh_auth_warndeprecate(
            "self.sshPrivateKey",
            "self._git_auth.ssh_private_key",
        )
        return self._git_auth.ssh_private_key

...

class GitStepMixin(GitMixin, _DeprecatedGitStepMixin):
```
with `_DeprecatedGitStepMixin` moved at the end of the file or even another file altogether.

Reply:
I think this makes code actually harder to understand. Just move these to the end of the file.

In any case, the properties are not documented as an API, so you can simply remove these.

Reply:
Since all that I marked with `ssh_auth_warndeprecate` are not documented, can I take this as a go on removing all of them?

Reply:
Yes.